tools/editors-libraries.adoc

```diff
@@ -0,0 +1,177 @@
+----
+<div id="new-dmn-editor-container"/>
+----
+
+Then load the Editor, passing the reference to the `div` container:
+
+[source,javascript,subs="+attributes"]
+----
+const newDmnEditor = NewDmnEditor.open({
+ container: document.getElementById("new-dmn-editor-container"), <1>
+ initialFileNormalizedPosixPathRelativeToTheWorkspaceRoot: "model.dmn" <2>
+ initialContent: Promise.resolve(""), <3>
+ readOnly: false, <4>
+ origin: "*", <5>
+ onError: () => {}, <6>
+ resources: new Map([]), <7>
+});
+----
+<1> `container`: HTML element to which the Editor will be appended to.
+<2> `initialContent`: Promise to the XML content of a DMN file for the DMN Editor. Can be empty. Examples:
+ * `Promise.resolve("")`
+ * `Promise.resolve("<DMN_CONTENT_DIRECTLY_HERE>")`
+ * `fetch("MyDecision.dmn").then(content => content.text())`
+<3> `initialFileNormalizedPosixPathRelativeToTheWorkspaceRoot`: The POSIX path relative to the workspace root (`/` by default), including the file name. The path is useful for locating the file in the workspace, especially in relation to other resources (defined by the `resources` parameters). Examples:
+ * `model.dmn`
+ * `myPath/model.dmn`
+ * `parent/child/filename.dmn`
+<4> `readOnly` (optional, defaults to `false`): Use `false` to allow content edition and `true` for read-only mode, in which the Editor will not allow changes.
+<5> `origin` (optional, defaults to `*` when accessing the application with the `file://` protocol, `window.location.origin` otherwise): If the application needs to change this parameter, use it.
+<6> `onError` (optional, defaults to `() => {}`): If there's an error opening the Editor, this function will be called.
+<7> `resources` (optional, defaults to an empty Map): Map of resources that will be provided for the Editor. This can be used, for instance, to provide included models for the DMN Editor. Each entry in the map has the resource POSIX path (similar to the `initialFileNormalizedPosixPathRelativeToTheWorkspaceRoot` parameter) as its key and an object containing the `content-type` (`text` or `binary`) and the resource `content` (Promise similar to the `initialContent` parameter) as its value. Resources located in a parent directory (in relation to the current content path, defined by the `initialFileNormalizedPosixPathRelativeToTheWorkspaceRoot`) won't be listed to be used as an Included Model. Examples:
+ * `resources: new Map([["types.dmn", { contentType: "text", content: Promise.resolve("<DMN_CONTENTS_HERE>")}], [...], [...]])`
+ * `resources: new Map([["/types/typeA.dmn", { contentType: "text", content: Promise.resolve("<DMN_CONTENTS_HERE>")}], ["/extras/decision.dmn", { contentType: "text", content: Promise.resolve("<DMN_CONTENTS_HERE>")}], [...]])`
+
+NOTE: The `initialFileNormalizedPosixPathRelativeToTheWorkspaceRoot` parameter is only available for the new DMN Editor. The classic editors don't have the concept of relative paths, so every mapped resource is available.
```

Comment:
The classic editors don't have the concept of relative paths, so every mapped resource is available -> The classic editors do not have the concept of relative paths and hence every mapped resource is available.